pkg/provider/provider.go

```diff
@@ -578,12 +578,9 @@ func (k *kubeProvider) Diff(
 	if tryDryRun {
 		patch, patchBase, err = k.dryRunPatch(oldInputs, newInputs)
 
-		// Fall back to input patch.
-		se, isStatusError := err.(*errors.StatusError)
-		if isStatusError && se.Status().Code == 400 &&
-			(se.Status().Message == "the dryRun alpha feature is disabled" ||
-				se.Status().Message == "the dryRun beta feature is disabled") {
-
+		// Fall back to input patch on error. Expected errors include dryRun being unsupported for a resource, or
+		// if the diff contains Output values.
+		if err != nil {
```

Comment:
Can we instead use `hasComputedValue` to look for unknown values in the new inputs and skip the dry-run entirely if it returns `true`?

Reply:
Yeah, we should not be doing this for every error.